### PR TITLE
Sovereign shard resolvers container factory

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Sovereign shard resolvers container factory
- 
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: dataRetriever/factory/resolverscontainer/interface.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package resolverscontainer
+
+import "github.com/multiversx/mx-chain-go/dataRetriever"
+
+type ShardResolversContainerFactoryCreator interface {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/interface.go:5]: Missing comment!

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package resolverscontainer
+
+import "github.com/multiversx/mx-chain-go/dataRetriever"
+
+type shardResolversContainerFactoryCreator struct {
+}
+
+func NewShardResolversContainerFactoryCreator() *shardResolversContainerFactoryCreator {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go:8]: Missing comments here and below!

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/multiversx/mx-chain-go/dataRetriever/resolvers"
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/process/factory"
+)
+
+type sovereignShardResolversContainerFactory struct {
+	*shardResolversContainerFactory
+}
+
+func NewSovereignShardResolversContainerFactory(shardContainer *shardResolversContainerFactory) (*sovereignShardResolversContainerFactory, error) {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go:16]: Missing comment!

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		return err
+	}
+
+	identifierHdr := factory.ExtendedHeaderProofTopic + shardC.CommunicationIdentifier(shardC.SelfId())
+	resolverSender, err := srcf.createOneResolverSenderWithSpecifiedNumRequests(identifierHdr, EmptyExcludePeersOnTopic, shardC.SelfId())
+	if err != nil {
+		return err
+	}
+
+	hdrNonceStore, err := srcf.store.GetStorer(dataRetriever.ExtendedShardHeadersNonceHashDataUnit)
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go:55]: hdrNonceStore**r**

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package resolverscontainer
+
+import "github.com/multiversx/mx-chain-go/dataRetriever"
+
+type sovereignShardResolversContainerFactoryCreator struct {
+}
+
+func NewSovereignShardResolversContainerFactoryCreator() *sovereignShardResolversContainerFactoryCreator {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go:8]: Missing comments here and below!

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+func TestNewShardResolversContainerFactory(t *testing.T) {
+	t.Parallel()
+
+	factory := resolverscontainer.NewShardResolversContainerFactoryCreator()
+	require.False(t, factory.IsInterfaceNil())
+	require.Implements(t, new(resolverscontainer.ShardResolversContainerFactoryCreator), factory)
+}
+
+func TestShardResolversContainerFactoryCreator_Create(t *testing.T) {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go:19]: _CreateShardResolversContainerFactory ?

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator_test.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+func TestSovereignNewShardResolversContainerFactory(t *testing.T) {
+	t.Parallel()
+
+	factory := resolverscontainer.NewSovereignShardResolversContainerFactoryCreator()
+	require.False(t, factory.IsInterfaceNil())
+	require.Implements(t, new(resolverscontainer.ShardResolversContainerFactoryCreator), factory)
+}
+
+func TestSovereignShardResolversContainerFactoryCreator_Create(t *testing.T) {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator_test.go:19]: _CreateShardResolversContainerFactory

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator_test.go [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/multiversx/mx-chain-go/dataRetriever/factory/resolverscontainer"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSovereignNewShardResolversContainerFactory(t *testing.T) {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator_test.go:11]: TestNewSovereignShardResolversContainerFactoryCreator ?

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/multiversx/mx-chain-go/dataRetriever/factory/resolverscontainer"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewShardResolversContainerFactory(t *testing.T) {
```
Comment (SebastianMarian) [inline dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go:11]: TestNewShardResolversContainerFactoryCreator ?

File: dataRetriever/factory/resolverscontainer/interface.go (hunk 1 above)
Comment (sasurobert) [inline dataRetriever/factory/resolverscontainer/interface.go:5]: comment

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go (hunk 2 above)
Comment (sasurobert) [inline dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go:8]: comment

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+	return &shardResolversContainerFactoryCreator{}
+}
+
+func (f *shardResolversContainerFactoryCreator) CreateShardResolversContainerFactory(args FactoryArgs,
```
Comment (sasurobert) [inline dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go:12]: comment

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go (hunk 3 above)
Comment (sasurobert) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go:16]: comment

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go (hunk 5 above)
Comment (sasurobert) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go:8]: comment

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+	return &sovereignShardResolversContainerFactoryCreator{}
+}
+
+func (f *sovereignShardResolversContainerFactoryCreator) CreateShardResolversContainerFactory(args FactoryArgs,
```
Comment (sasurobert) [inline dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go:12]: comment

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go (hunk 5 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go:8]: Fixed

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go (hunk 6 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go:19]: Right, fixed it

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go (hunk 9 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator_test.go:11]: Right, fixed it

File: dataRetriever/factory/resolverscontainer/interface.go (hunk 1 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/interface.go:5]: Fixed it

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go (hunk 2 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go:8]: Fixed it

File: dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go (hunk 10 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/shardResolversContainerFactoryCreator.go:12]: Fixed it

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go (hunk 3 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go:16]: Fixed it

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go (hunk 5 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go:8]: Fixed it

File: dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go (hunk 11 above)
Comment (mariusmihaic) [reply dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactoryCreator.go:12]: Fixed it

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**